lightning/src/onion_message/messenger.rs

```diff
@@ -933,7 +933,18 @@ where
 		Ok((Payload::Receive::<ParsedOnionMessageContents<<<CMH as Deref>::Target as CustomOnionMessageHandler>::CustomMessage>> {
 			message, control_tlvs: ReceiveControlTlvs::Unblinded(ReceiveTlvs { path_id }), reply_path,
 		}, None)) => {
-			Ok(PeeledOnion::Receive(message, path_id, reply_path))
+			match (&message, &path_id) {
+				(_, None) => {
+					Ok(PeeledOnion::Receive(message, None, reply_path))
+				}
+				(ParsedOnionMessageContents::Offers(_), Some(RecipientData::OffersContext(_))) => {
+					Ok(PeeledOnion::Receive(message, path_id, reply_path))
+				}
+				(ParsedOnionMessageContents::Custom(_), Some(RecipientData::CustomContext(_))) => {
+					Ok(PeeledOnion::Receive(message, path_id, reply_path))
+				}
+				_ => Err(())
```

Comment:
Let's also `log_trace` indicating the message was sent on a blinded path with the wrong context. Ideally, indicating the message and context types. Not sure if we want to include the `Debug` content given this would typically be a result of malicious behavior.

Reply:
Introduce a `log_trace!()` for the wrong context in [e4d4218](https://github.com/lightningdevkit/rust-lightning/commit/e4d4218954dbe9564ed1f2ded56f7af515e2950f).

Including `Debug` content could be relevant if the issue arose from us creating a blinded path with the wrong context when we initially provided our peer with a blinded path to respond with. However, since this issue can also result from malicious or faulty peer behavior, assuming our fault through a `debug_assert!()` might not be the best choice in this case.